src/azure-cli/azure/cli/command_modules/storage/_client_factory.py

```diff
@@ -204,7 +204,13 @@ def cf_blob_service(cli_ctx, kwargs):
     token_credential = kwargs.pop('token_credential', None)
     sas_token = kwargs.pop('sas_token', None)
     if connection_string:
-        return t_blob_service.from_connection_string(conn_str=connection_string)
+        try:
+            return t_blob_service.from_connection_string(conn_str=connection_string)
+        except ValueError:
+            from azure.cli.core.azclierror import InvalidArgumentValueError
+            raise InvalidArgumentValueError('Invalid connection string',
```

Comment:
Can we also show the invaid connection string in error message?

Reply:
Done